formver.compiler-plugin/uniqueness/src/org/jetbrains/kotlin/formver/uniqueness/UniqueCheckerContext.kt

```diff
@@ -5,15 +5,36 @@
 
 package org.jetbrains.kotlin.formver.uniqueness
 
+import org.jetbrains.kotlin.fir.FirAnnotationContainer
 import org.jetbrains.kotlin.fir.FirSession
-import org.jetbrains.kotlin.fir.declarations.FirDeclaration
+import org.jetbrains.kotlin.fir.declarations.hasAnnotation
+import org.jetbrains.kotlin.fir.symbols.FirBasedSymbol
 import org.jetbrains.kotlin.formver.common.ErrorCollector
 import org.jetbrains.kotlin.formver.common.PluginConfiguration
+import org.jetbrains.kotlin.name.ClassId
+
+sealed class HasAnnotation {
+    class Symbol(val symbol: FirBasedSymbol<*>) : HasAnnotation()
+    class AnnotationContainer(val container: FirAnnotationContainer) : HasAnnotation()
+
+    fun hasAnnotation(id: ClassId, session: FirSession): Boolean {
+        return when (this) {
+            is Symbol -> symbol.hasAnnotation(id, session)
+            is AnnotationContainer -> container.hasAnnotation(id, session)
+        }
+    }
```

Comment:
I think this would make more sense to handle with a virtual call, since you're defining it right here and are dispatching on precisely that information.